lib/ansible/modules/cloud/azure/azure_rm_eventhubconsumergroup.py

```diff
@@ -0,0 +1,227 @@
+try:
+    from msrestazure.tools import parse_resource_id
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+class AzureRMEventHubConsumerGroup(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            namespace=dict(
+                type='str',
+                required=True
+            ),
+            eventhub=dict(
+                type='str',
+                required=True
+            ),
+            consumer_group=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            user_metadata=dict(
+                tyep='str'
+            )
+        )
+
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        self.resource_group = None
+        self.namespace = None
+        self.eventhub = None
+        self.consumer_group = None
+        self.state = None
+        self.user_metadata = None
+
+        super(AzureRMEventHubConsumerGroup, self).__init__(self.module_arg_spec, supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+        
+        changed = False
+        result = None
+
+        result = self.get_consumer_group()
+        if self.state == 'present':
+            if not result:
+                changed = True
+                self.log('Creating a consumer group')
+                if not self.check_mode:
+                    result = self.create_or_update_consumer_group().as_dict()
+            else:
+                # Compare user_metadata
+                if self.user_metadata and self.user_metadata != result.user_metadata:
+                    changed = True
+                
+                if changed and not self.check_mode:
+                    result = self.create_or_update_consumer_group().as_dict()
+        elif result:
+            changed = True
+            if not self.check_mode:
+                self.delete_consumer_group()
+            result = True
+        self.results['changed'] = changed
+        self.results['state'] = result
+        return self.results
+
+    def get_consumer_group(self):
+        try:
+            return self.eventhub_client.consumer_groups.get(self.resource_group, self.namespace, self.eventhub, self.consumer_group)
+        except Exception as exc:
+            pass
+            return None
+
+    def create_or_update_consumer_group(self):
+        try:
+            return self.eventhub_client.consumer_groups.create_or_update(self.resource_group,
```

Comment:
again, avoid positional paramaeters, but specific parameters name

Reply:
Thank you very much! I will modify the codes based on the comments